annotator/annotator.py

```diff
@@ -356,10 +392,13 @@ def __init__(self, start, end, doc, label=None):
 
     def overlaps(self, other_span):
         return (
-            (self.start >= other_span.start and self.start <= other_span.end) or
-            (other_span.start >= self.start and other_span.start <= self.end)
+            (self.start >= other_span.start and self.start < other_span.end) or
+            (other_span.start >= self.start and other_span.start < self.end)
```

Comment:
There was a bug before. There is no overlap when the end index is the same as a start index. The span includes only the characters up to the end index, but not the character at the end index.